lib/framework/sync_job_runner.rb

```diff
@@ -20,7 +20,7 @@ def initialize(expected_fields, actual_fields)
   class SyncJobRunner
     def initialize(connector_settings)
       @connector_settings = connector_settings
-      @connector_instance = Connectors::REGISTRY.connector(@connector_settings.service_type)
+      @connector_instance = Connectors::REGISTRY.connector(App::Config['service_type'])
```

Comment:
Kibana won't populate `service_type` in connector configs. We need to rely on `service_type` in yaml file.
cc @sphilipse @artem-shelkovnikov @mchernyavskaya

Reply:
TBH I think [this](https://github.com/elastic/connectors/pull/149/files#diff-8c4301d73e050dc0f0b752430187d390d0609064f769ad6cdf879bda29ffb29a) place is a tiny bit better as we're already using `App::Config` there and don't add extra imports.

Reply:
I don't use `require` here and it also worked, and I can add it. 

I tend to avoid passing app-wide accessible resources around.

Reply:
I don't think App::Config is an app-wide accessible resource - it can be used by everything in `App` namespace, but every other module - `core`, `utils` or `connectors` should not have direct access to it, at least for now.

I think we need to have a separate discussion about singletons/app wide abstractions and what is provided where, it's a solid and major topic to discuss

Reply:
> but every other module - `core`, `utils` or `connectors` should not have direct access to it, at least for now.

Half of me is convinced, e.g. A `util` should work without dependency on app-level configs. But our `EsClient` reads configs directly (This maybe wrong). I think we should have our 2nd engineers sync soon.